src/qibolab/channel.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class NamedChannel:
+    """Channel that has a name. This is part of the end-user API, i.e. is in
+    the top layer.
+
+    The idea is the following:
+        1. End users do not know what are the types of channels in a platform, they just know names.
+           They use only NamedChannels to describe pulse sequences. They can create NamedChannels using the get_channel function below.
+        2. User makes some assumptions about the types of channels. E.g. they assume that the channel NamedChannel("qubit_0/drive")
+           is an IQ channel, hence they play IQ pulses on it, and they provide IQChannelConfig for it.
+        3. Upon receival of the execution request qibolab validates that the requested execution can be done, i.e.
+           channels with those names exist and user's assumptions are correct.
+        4. qibolab proceeds with execution.
+    For the last two steps qibolab needs to replace generic NamedChannels with concrete channels (e.g. ZurichIQChannel, QbloxDCChannel, etc.), and
+    those should be available in the Platform description.
+
+    TODO: I am not sure if having this class brings any benefit to this plan compared to the case where we just use naked str names, but I will figure
+    this out later during implementation.
+
+    TODO: One might argue that it is reasonable to provide the end user the types of channels as well, and then do all the validation while constructing the pulse
+    sequence. I thought about this and failed to find real benefit, it just seems to complicate the code and the user-facing API for no real benefit.
+    Please comment if you have anything to say regarding this.
```

Comment:
In case you want to benefit from types, but your underlying data is just a `str`, avoid making a full-blown `class` for that, and make a type alias:
```py
NamedChannel = str
```

If you wish to have well-distinguished types, and to avoid that every `str` could be seen as a `NamedChannel` and the opposite (without an explicit cast), adopt the [`NewType`](https://docs.python.org/3/library/typing.html#typing.NewType) pattern.
```py
from typing import NewType

NamedChannel = NewType('NamedChannel', str)
```